include/clap/factory/draft/preset-discovery.h

```diff
@@ -0,0 +1,315 @@
+                             uint32_t                                              flags);
+
+   // Sets the preset name.
+   // The preset name can be infered from the file name, but it is mandatory to set it for
+   // container's preset.
+   void(CLAP_ABI *set_name)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                            const char                                           *name);
+
+   // Adds a creator name for the preset.
+   void(CLAP_ABI *add_creator)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                               const char                                           *creator);
+
+   // Sets a description of the preset.
+   void(CLAP_ABI *set_description)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                                   const char *description);
+
+   // Sets the creation time and last modification time of the preset.
+   // The timestamps are in seconds since EPOCH.
+   // If one of the time isn't known, then set it to 0.
+   // If this function is not called, then the indexer may look at the file's creation and
+   // modification time.
+   void(CLAP_ABI *set_timestamps)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                                  uint64_t creation_time,
+                                  uint64_t modification_time);
+
+   // Adds a feature to the preset.
+   //
+   // The feature string is arbitrary, it is the indexer's job to understand it and remap it to its
+   // internal categorization and tagging system.
+   //
+   // Examples:
+   // kick, drum, tom, snare, clap, cymbal, bass, lead, metalic, hardsync, crossmod, acid,
+   // distorted, drone, pad, dirty, etc...
+   void(CLAP_ABI *add_feature)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                               const char                                           *feature);
+
+   // Adds extra information to the metadata.
+   void(CLAP_ABI *add_extra_info)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                                  const char                                           *key,
+                                  const char                                           *value);
+} clap_preset_discovery_metadata_receiver_t;
+
+typedef struct clap_preset_discovery_filetype {
+   const char *name;
+   const char *description;
+
+   // `.' isn't included in the string.
+   // If empty or NULL then every file should be matched.
+   const char *file_extension;
+} clap_preset_discovery_filetype_t;
+
+// Defines a place in which to search for presets
+typedef struct clap_preset_discovery_location {
+   uint32_t    flags; // see enum clap_preset_discovery_flags
+   const char *name;  // name of this location
+
+   // URI
+   // - file:/// for pointing to a file or directory; directories are scanned recursively
+   //   eg: file:///home/abique/.u-he/Diva/presets/
```

Comment:
It may also be a good idea to list an example of a path on Windows here since things work a little differently there, and different hosts/plugins may end up doing different things otherwise. So this would become `file://C:/Users/abique/<whatever the equivalent path is on Windows>`.

Reply:
https://en.wikipedia.org/wiki/File_URI_scheme says the Shell API works like this (file:/// rather than file://):

file:///c:/Users/abique/.u-he/Diva/presets/

Reply:
Fixed.